Jsondump

DanielMorsing opened the pull request:
This adds support for writing a JSON representation of capnproto structs and enums.

For most users, the MarshalJSON implementation will be enough but for large structs, there's the WriteJSON method which will stream the JSON data into a writer.

Commit:
Add a simple method for marshaling to json.

This adds a simple json dumper to the generated files.

Commit:
Add MarshalJSON forwarder

Commit:
Don't import JSON imports when file has no JSON writing types.

Commit:
Handle TEXT_DATA and lists.

This changes the codegen so that every field get its own block with a redeclaration. Keeps the code handling names and values separate.

DanielMorsing commented:
I just realised that this has garbage author/commit ids. Should I go fix that?

jmckaskill commented:
Yes

-- James

On Wed, Jan 22, 2014 at 12:20 PM, Daniel Morsing
notifications@github.comwrote:

> I just realised that this has garbage author/commit ids. Should I go fix
> that?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jmckaskill/go-capnproto/pull/8#issuecomment-33045198
> .

DanielMorsing commented:
OK, Done.

Sorry about that.

Commit:
Don't declare encoder if not used.

If the encoder wasn't used in a function, the Go compiler would not compile the file.

Commit:
Add error checking on JSON formatting

Commit:
Add missing error check.

alberts commented:
@DanielMorsing it would be nice to fix the JSON encoding to not use the Encoder so that we can get rid of the newlines in the output.

Commit:
Use json.Marshal instead of encoder

Gets rid of some of the nastier pieces of logic in the compiler and doesn't put a newline on the end of each primitive value.

alberts commented:
@jmckaskill I've tested this patch with one of our bigger .capnp files and it works well. Any other comments for us?
